src/compiler/diagnosticMessages.json

```diff
@@ -895,7 +895,7 @@
         "category": "Error",
         "code": 1322
     },
-    "Dynamic import cannot be used when targeting ECMAScript 2015 modules.": {
+    "Dynamic import cannot be used when targeting ECMAScript 2015 modules. Please use 'esnext' or 'commonjs' as module compiler option.": {

```

Comment:
how about `Dynamic import is only supported when '--module' flag is 'commonjs' or `esNext`.